src/Connection/Http1Connection.php

```diff
@@ -332,26 +368,44 @@ private function normalizeTraceRequest(Request $request): Request
         return $request;
     }
 
-    private function shouldCloseSocketAfterResponse(Response $response): bool
+    private function determineKeepAliveTimeout(Response $response): int
     {
         $request = $response->getRequest();
 
         $requestConnHeader = $request->getHeader('connection');
         $responseConnHeader = $response->getHeader('connection');
 
-        if ($requestConnHeader && !\strcasecmp($requestConnHeader, 'close')) {
-            return true;
+        if (!\strcasecmp($requestConnHeader, 'close')) {
+            return 0;
+        }
+
+        if ($response->getProtocolVersion() === '1.0') {
+            return 0;
+        }
+
+        if (\strcasecmp($responseConnHeader, 'keep-alive')) {
```

Comment:
Should be checking against `close`, too, as `keep-alive` is the default for HTTP/1.1.

Reply:
If the value is anything other than `'keep-alive'` the connection will be closed.